### PR TITLE
Formula yourls

### PR DESCRIPTION
**Pull request by netomarin**
**- What I did**
Added two formulas to use a self-hosted YOURLS instance (https://yourls.org/) to shortener your own URLs without depending on public services, like bit.ly.

**- How I did it**
`yourls init`: Shell script formula to create a local file with API endpoint and secret.
`yourls shorturl`: Java formula to short a given url

**- How to verify it**
It's required to have a YOURLS instance and get the API endpoint and a valid secret. With those attributes, just invoke `rit yourls init` to enter configuration values, and then `rit yourls shorturl` and provide the requested parameters to short an url.

Let me know if you have any problem running it, I can provide some screenshots and a short video.

**- Description for the changelog**

Added the group yourls and two formulas to shorten urls using a self hosted YOURLS server.

### CODE REVIEW

**Comment by fabianofernandeszup** [comment]
Hello @netomarin , first thanks for your contribution. We don't really have a formula like yours yet, so I'm sure it will be very useful for everyone in the community.

I made some comments in your PR to improve the user experience. Feel free to talk about it.

I would also like to inform you that we are almost launching version 2 with several improvements and evolutions. As soon as it is released I would like to propose this formula to you.

File: yourls/shorturl/config.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    {
+      "name" : "url_to_shorten",
+      "type" : "text",
+      "label" : "Enter the url to shorten : "
+    },
+    {
+      "name" : "keyword",
+      "type" : "text",
+      "label" : "Short url keyword [OPTIONAL]"
+    }
```
Comment (fabianofernandeszup) [inline yourls/shorturl/config.json:13]: For a better user experience, use ritchie's CREDENTIALS feature to store and load data like this. I will show an example in the cofig.json file.

```
    {
      "name" : "YOURLS_API_ENDPOINT",
      "label" : "CREDENTIAL_YOURLS_APIENDPOINT"
    },
    {
      "name" : "YOURLS_API_SECRET",
      "label" : "CREDENTIAL_YOURLS_APISECRET"
    }
```
**Name pattern**: String without spaces
**Label pattern**: 3 uppercase words used by undescore without spaces, 
starting with the word **CREDENTIAL**, followed by the name of the provider 
created **YOURLS** and finally the defined key **APIENDPOINT** or **APISECRET** without spaces or separations 
with a hyphen or underscore

Example of creating new credentials:

criando-provider-yourul

File: yourls/shorturl/src/Main.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import shorturl.Shorturl;
+
+public class Main {
+
+    public static void main(String[] args) throws Exception {
+        String input1 = System.getenv("URL_TO_SHORTEN");
+        String input2 = System.getenv("KEYWORD");
```
Comment (fabianofernandeszup) [inline yourls/shorturl/src/Main.java:7]: Using credentials you can retrieve them that way.
```
String apiEndPoint = System.getenv("YOURLS_API_ENDPOINT");
String apiSecret = System.getenv("YOURLS_API_SECRET");
```

File: yourls/shorturl/src/shorturl/Shorturl.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+    private String urlToShorten;
+    private String keyword;
+
+    public void Run() throws Exception {
+
+        //Load API config
+        Properties prop = null;
+        try {
+            prop = readYOURLSProperties();
```
Comment (fabianofernandeszup) [inline yourls/shorturl/src/shorturl/Shorturl.java:31]: With the use of credentials, this verification would not be necessary, as the **ritchie** checks whether they are set according to config.json and returns an error to the user who executed the formula.

File: yourls/init/src/init/init.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+run() {
+  #removing previous configuration if exists
+  rm -rf $HOME/.yourls
+  #(re)creating local directory to store configuration information
+  mkdir $HOME/.yourls
+  echo "yourls_api_endpoint = $YOURLS_API_ENDPOINT" > $HOME/.yourls/yourls.properties
+  echo "yourls_api_secret = $YOURLS_API_SECRET" >> $HOME/.yourls/yourls.properties
```
Comment (fabianofernandeszup) [inline yourls/init/src/init/init.sh:8]: This formula can be replaced by the CREDENTIALS feature of the ritchie. Simplifying its use.

**Comment by netomarin** [comment]
> I made some comments in your PR to improve the user experience. Feel free to talk about it.

I'm not seeing the changes requested, can you confirm you have sent them?

> would also like to inform you that we are almost launching version 2 with several improvements and evolutions. As soon as it is released I would like to propose this formula to you.

Sure! Glad to help.

**Pull request merged**